spot_driver_cpp/include/spot_driver_cpp/api/kinematic_api.hpp

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2023 Boston Dynamics AI Institute LLC. All rights reserved.
+
+#pragma once
+
+#include <bosdyn/client/inverse_kinematics/inverse_kinematics_client.h>
+#include <bosdyn/client/service_client/result.h>
+
+#include <tl_expected/expected.hpp>
+
+#include <string>
+
+namespace spot_ros2 {
+
+using ::bosdyn::api::spot::InverseKinematicsRequest;
+using ::bosdyn::api::spot::InverseKinematicsResponse;
+using ::bosdyn::client::Result;
+
+class KinematicApi {
+ public:
+  /**
+   * Return a solution to the given request.
+   */
+  virtual tl::expected<Result<InverseKinematicsResponse>, std::string> get_solutions(
+      InverseKinematicsRequest& request) = 0;
```

Comment:
Since you modified `CPPLINT.cfg` this stands out to me -- why does `request` need to be a non-const reference?

Reply:
When you spotted, this I thought it was an error, but then I found out that this method passes the `InverseKinematicsRequest` into a method of `bosdyn::client::InverseKinematicsClient` which requires a non-const ref. I assume the inner method modifies the request.
```
kinematic_client_->InverseKinematics(request);
```

The main reason I modified the CPPLINT.cfg is because it was incorrectly asking me to replace non-const ref in the converter methods with a raw pointer. This is incorrect because a raw pointer can be set to nullptr, while in the converter method source and destination objects are both required.